File: AndroidClient/LdpAndroidClient/app/src/main/java/com/ldp/androidclient/network/ILdpObservable.java
package com.ldp.androidclient.network;


import com.ldp.androidclient.protocol.*;

public interface ILdpObservable {
    void addListener(ILdpPacketHandler listener);
    void removeListener(ILdpPacketHandler listener);
    void removeListeners();
    void notifyToAllListeners(LdpProtocol.LdpPacket packet);
}

File: Server/Server/Network/Handlers/PacketHandlerBase/ILdpObservable.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Server.Network.Handlers.PacketHandlerBase
{
    interface ILdpObservable
    {
        void AddListener(ILdpPacketHandler listener);
        void RemoveListener(ILdpPacketHandler listener);
        void RemoveListeners();
        void NotifyToAllListeners(LdpPacket packet);
    }
}
